src/lavinmq/amqp/queue/message_store.cr

```diff
@@ -347,6 +347,9 @@ module LavinMQ
                 @segments.delete seg
                 next
               end
+            rescue ex
+              Log.error { "Closing message store: invalid SchemaVersion in #{path}" }
```

Comment:
I think the log message could be even more descriptive, something like "Could not initialize segment, closing message store: Invalid SchemaVersion in #{path}"